src/mnt_setup_fnc.cpp

```diff
@@ -8,59 +8,88 @@
 std::string const DATADIR = "/storage/epp2/phshgg/DVTuples__v23/";
 
 struct path_data {
-	std::string const DATADIR;
-	std::string sqrts,
+	std::string const DATADIR,
+	sqrts,
 	stripping,
 	year,
 	polarity,
 	stream;
 };
 
-void make_histogram(struct path_data path_in, std::string hist_name, std::string hist_setup, int const nbins, float const xmin, float const xmax, std::string expression, THStack *hist_stack) {
+struct input {
+	std::string const expression;
+	int const nbins;
+	float const xmin,
+	xmax;
+	std::string const label,
+	unit;
+};
+
+TH1F* make_TH1F(struct path_data path_in, struct input hist_input, std::string hist_name, std::string hist_setup) {
 	TChain ch("Z/DecayTree");
 	std::string path = path_in.DATADIR + path_in.sqrts + "TeV_" + path_in.year + "_" + path_in.stripping + "_" + path_in.polarity + "_" + path_in.stream + ".root";
 	ch.Add(path.c_str());
 	
-	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), nbins, xmin, xmax);
-	expression += hist->GetName();
+	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), hist_input.nbins, hist_input.xmin, hist_input.xmax);
+	std::string expression = hist_input.expression +">>"+ hist->GetName();
 	ch.Draw(expression.c_str());
-	hist_stack->Add(hist);
+	return hist;
 }
 
+void plot_histogram(struct path_data mnt_in, struct path_data sim_in, struct input hist_input) {

```

Comment:
Could the name of this function be more clearly related to its purpose?
I think the purpose is to compare data and simulation for some distribution. 
E.g. `plot_data_sim`?